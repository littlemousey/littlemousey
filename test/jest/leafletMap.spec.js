import { shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import leafletMap from '@/components/leafletMap'

Vue.use(Vuetify)

let wrapper

beforeEach(() => {
  wrapper = shallowMount(leafletMap, {
    propsData: {
      markers: [{ city: 'Jakarta', coordinates: [-6.17511, 106.865036] }]
    },
    computed: {
      resizeMap() {
        return 'height: 300px; width: 1000px'
      }
    },
    stubs: {
      LMap: true,
      LTileLayer: true,
      LMarker: true,
      LPopup: true,
      'v-marker-cluster': true,
      'no-ssr': true
    }
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('Leaflet map', () => {
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
