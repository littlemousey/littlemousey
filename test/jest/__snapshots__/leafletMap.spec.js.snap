// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaflet map renders correctly 1`] = `
<no-ssr-stub>
  <lmap-stub
    center="50,20"
    crs="[object Object]"
    options="[object Object]"
    style="height: 300px; width: 1000px;"
    worldcopyjump="true"
    zoom="2"
  >
    <ltilelayer-stub
      opacity="1"
      options="[object Object]"
      pane="tilePane"
      tilelayerclass="function tileLayer(url, options) {
	return new TileLayer(url, options);
}"
      tilesize="256"
      url="http://{s}.tile.osm.org/{z}/{x}/{y}.png"
      visible="true"
      zindex="1"
    />
     
    <v-marker-cluster-stub
      options="[object Object]"
    >
      <lmarker-stub
        icon="[object Object]"
        latlng="-6.17511,106.865036"
        options="[object Object]"
        pane="markerPane"
        visible="true"
      >
        <lpopup-stub
          latlng=""
          options="[object Object]"
        >
          Jakarta
        </lpopup-stub>
      </lmarker-stub>
    </v-marker-cluster-stub>
  </lmap-stub>
</no-ssr-stub>
`;
